src/main/java/pl/cleankod/exchange/entrypoint/AccountController.java

```diff
@@ -1,59 +1,89 @@
-                )
-                .orElseGet(() ->
-                        findAccountUseCase.execute(Account.Id.of(id))
-                                .map(ResponseEntity::ok)
-                                .orElse(ResponseEntity.notFound().build())
-                );
+    @ResponseStatus(HttpStatus.OK)
+    public AccountResponse findAccountById(@RequestAttribute("CID") String cid,
+                                           @ApiParam(example = "fa07c538-8ce4-11ec-9ad5-4f5a625cd744", required = true) @PathVariable String id,
+                                           @ApiParam(example = "EUR") @RequestParam(required = false) String currency) {
+        Preconditions.validateUUIDFormat(id);
+        Preconditions.validateCurrency(currency);
+        return accountService.findAccountBy(Account.Id.of(UUID.fromString(id)), currency, cid);
     }
 
+    @ApiOperation
+            (
+                    value = "Returns an account based on an account number",
+                    httpMethod = "GET"
+            )
+    @ApiResponses(value = {
+            @ApiResponse(code = 200,
+                    message = "",
+                    response = AccountResponse.class),
+            @ApiResponse(code = 404,
+                    message = "The requested account was not found.",
+                    response = ResponseEntity.class),
+            @ApiResponse(code = 400,
+                    message = "Bad request",
+                    response = ResponseEntity.class)
+    }
+    )
     @GetMapping(path = "/number={number}")
-    public ResponseEntity<Account> findAccountByNumber(@PathVariable String number, @RequestParam(required = false) String currency) {
+    @ResponseStatus(HttpStatus.OK)
+    public AccountResponse findAccountByNumber(@RequestAttribute("CID") String cid,
+                                               @ApiParam(example = "65+1090+1665+0000+0001+0373+7343", required = true) @PathVariable String number,
+                                               @ApiParam(example = "PLN") @RequestParam(required = false) String currency) {
+
         Account.Number accountNumber = Account.Number.of(URLDecoder.decode(number, StandardCharsets.UTF_8));
-        return Optional.ofNullable(currency)
-                .map(s ->
-                        findAccountAndConvertCurrencyUseCase.execute(accountNumber, Currency.getInstance(s))
-                                .map(ResponseEntity::ok)
-                                .orElse(ResponseEntity.notFound().build())
-                )
-                .orElseGet(() ->
-                        findAccountUseCase.execute(accountNumber)
-                                .map(ResponseEntity::ok)
-                                .orElse(ResponseEntity.notFound().build())
-                );
+        Preconditions.validateAccountNumber(accountNumber.value());
+        Preconditions.validateCurrency(currency);
+        return accountService.findAccountBy(accountNumber, currency, cid);
```

Comment:
Do you think that the `correctionId` would generally be sent as a request param?